internal/version.go

```diff
@@ -39,15 +38,21 @@ func detectECKVersion(c *kubernetes.Clientset, namespace, userSpecifiedVersion s
 		}
 		return parsed
 	}
-	statefulSet, err := c.AppsV1().StatefulSets(namespace).Get(context.Background(), "elastic-operator", metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return extractVersionFromDeployment(c, namespace)
-	}
+	// we use the control-plane label since ECK version 1.0
+	ssets, err := c.AppsV1().StatefulSets(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "control-plane=elastic-operator"})
```

Comment:
Shouldn't we also do the same with a label selector in extractVersionFromDeployment ()?

Reply:
We don't have those labels on the Deployment if it is generated by OLM only on the individual Pods